client/X11/xf_cliprdr.c

```diff
@@ -1663,6 +1928,334 @@ static UINT xf_cliprdr_clipboard_file_range_failure(wClipboardDelegate* delegate
 	return clipboard->context->ClientFileContentsResponse(clipboard->context, &response);
 }
 
+static void xf_cliprdr_fuse_getattr(fuse_req_t req, fuse_ino_t ino, struct fuse_file_info* fi)
+{
+	double timeout = 0;
+	struct stat stbuf;
+
+	xfCliprdrFuseInode* node;
+	xfClipboard* clipboard = (xfClipboard*)fuse_req_userdata(req);
+
+	ArrayList_Lock(clipboard->ino_list);
+	node = (xfCliprdrFuseInode*)ArrayList_GetItem(clipboard->ino_list, ino - 1);
+	ArrayList_Unlock(clipboard->ino_list);
+	if (!node)
+	{
+		fuse_reply_err(req, ENOENT);
+	}
+	else
+	{
+		memset(&stbuf, 0, sizeof(stbuf));
+		stbuf.st_ino = ino;
+		stbuf.st_mode = node->st_mode;
+		stbuf.st_mtime = node->st_mtim.tv_sec;
+		if (ino == FUSE_ROOT_ID)
+		{
+			stbuf.st_nlink = 2;
+		}
+		else
+		{
+			stbuf.st_nlink = 1;
+		}
```

Comment:
Ternary operator makes this a bit more compact:

```c
stbuf.st_nlink = (ino == FUSE_ROOT_ID) ? 2 : 1;
```

Though, is this required for correctness? Why can't we just say "1 link" for everything?

I mean, strictly speaking, each regular file should have 1 link while for directories it depends on the number of subdirectories in them. Getting that number accurate is a bother and is not particularly important, I think.